apps/api_web/lib/api_web/controllers/live_facility_controller.ex

```diff
@@ -35,29 +35,40 @@ defmodule ApiWeb.LiveFacilityController do
     response(429, "Too Many Requests", Schema.ref(:TooManyRequests))
   end
 
-  def index_data(conn, params) do
-    with {:ok, filtered} <- Params.filter_params(params, @filters, conn),
-         {:ok, _includes} <- Params.validate_includes(params, @includes, conn) do
-      case filtered do
-        %{"id" => ids} ->
-          ids
-          |> split_on_comma
-          |> State.Facility.Parking.by_facility_ids()
-          |> Enum.group_by(& &1.facility_id)
-          |> Enum.map(fn {facilty_id, properties} ->
-            %{
-              facility_id: facilty_id,
-              properties: properties,
-              updated_at: updated_at(properties)
-            }
-          end)
-          |> State.all(Params.filter_opts(params, @pagination_opts))
+  def check_version(conn) do
```

Comment:
You can probably implement this as a Plug, something like:
```elixir
# at the top
plug :ensure_path_matches version

def ensure_path_matches_version(conn, _) do
  if matches? do
    conn
  else
    conn
    |> do_404
    |> halt
  end
end
```

The `halt/1` will prevent any other processing from happening in the controller, and so the `index_data` and `show_data` function don't need to worry about that.

Reply:
Great idea! Thanks for letting me know. Refactored.